packages/kbn-discover-utils/src/data_types/logs/logs_context_service.ts

```diff
@@ -28,15 +27,32 @@ export const DEFAULT_ALLOWED_LOGS_BASE_PATTERNS = [
   'winlogbeat',
 ];
 
-export const createLogsContextService = (_deps: LogsContextServiceDeps = {}) => {
-  // This is initially an hard-coded set of well-known base patterns,
-  // we can extend this allowed list with any setting coming from uiSettings
-  const ALLOWED_LOGS_DATA_SOURCES = [createRegExpPatternFrom(DEFAULT_ALLOWED_LOGS_BASE_PATTERNS)];
+export const createLogsContextService = async ({
+  logsDataAccessPlugin,
+}: LogsContextServiceDeps) => {
+  let logSources;
 
+  if (logsDataAccessPlugin) {
+    const logSourcesService = logsDataAccessPlugin.services.logSourcesService;
+    logSources = (await logSourcesService.getLogSources())
+      .map((logSource) => logSource.indexPattern)
+      .join(',') // TODO: Will be replaced by helper in: https://github.com/elastic/kibana/pull/192003
+      .split(',');
+  }
+
+  const ALLOWED_LOGS_DATA_SOURCES = [
+    createRegExpPatternFrom(DEFAULT_ALLOWED_LOGS_BASE_PATTERNS),
+    ...(logSources ? logSources : []),
+  ];
+
+  return getService(ALLOWED_LOGS_DATA_SOURCES);
```

Comment:
ℹ️ This additional function just makes testing / mocking a bit easier, so we don't have to worry about the "wrapper" being async.